.circleci/config.yml

```diff
@@ -0,0 +1,122 @@
+version: 2.1
+
+orbs:
+  os-detect: circleci/os-detect@0.2
+
+executors:
+  linux_arm64:
+    description: "arm64"
+    machine:
+      image: ubuntu-2004:202101-01
+    resource_class: arm.medium
+  linux_amd64:
+    description: "amd64"
+    machine:
+      image: ubuntu-2004:202101-01
+    resource_class: medium
+
+commands:
+  install_go_linux:
+    parameters:
+      arch:
+        default: "amd64"
+        type: string
+      version:
+        type: string
+    steps:
+      - os-detect/init # this setup the '$SUDO' variable
+      - run:
+          name: "install Golang linux"
+          command: |
+            if command -v go >/dev/null; then
+              if go version | grep -q -F "go<< parameters.version >> "; then
+                echo "Binary already exists, skipping download."
+                exit 0
+              fi
+              echo "Error different version of Go already installed: '`go version`' when requested was '<< parameters.version >>'"
+
+              $SUDO rm -rf /usr/local/go
+              $SUDO install --owner=${USER} -d /usr/local/go
+            fi
+
+            echo "Installing the requested version of Go."
+
+            curl --fail --location -sS "https://dl.google.com/go/go<<parameters.version >>.linux-<< parameters.arch >>.tar.gz" \
+            | sudo tar --no-same-owner --strip-components=1 --gunzip -x -C /usr/local/go/
+
+            echo "export PATH=$PATH:/usr/local/go/bin" >> $BASH_ENV
+            $SUDO chown -R $(whoami): /usr/local/go
```

Comment:
Nit: Would be nice to have all params with vars enclosed in quotes to prevent unwanted token splitting. I know we control most of them but it's good practice to use on shell scripts.